sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs

```diff
@@ -143,13 +143,20 @@ private static ValueTask ProcessNextAsync(HttpMessage message, ReadOnlyMemory<Ht
             if (currentActivity != null)
             {
                 var currentActivityId = currentActivity.Id ?? string.Empty;
-
+#if NETCOREAPP2_1
                 if (currentActivity.IsW3CFormat())
+#else
+                if (currentActivity.IdFormat == ActivityIdFormat.W3C)
+#endif
                 {
                     if (!message.Request.Headers.Contains(TraceParentHeaderName))
                     {
                         message.Request.Headers.Add(TraceParentHeaderName, currentActivityId);
+#if NETCOREAPP2_1
                         if (currentActivity.GetTraceState() is string traceStateString)
```

Comment:
Will we ever be able to remove this? If not, I'm not sure it is worth adding the conditional.

Reply:
Ah nevermind, this at least makes us AOT compliant for all targets above netcoreapp2.1